src/components/DoughnutChart/DoughnutChart.test.tsx

```diff
@@ -0,0 +1,86 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import React from "react";
+import DoughnutChart, { TestIds } from "./DoughnutChart";
+import userEvent from "@testing-library/user-event";
+
+describe("DoughnutChart", () => {
+  const defaultProps = {
+    chartID: "test",
+    segmentHoverWidth: 10,
+    segmentWidth: 8,
+    size: 100,
+    segments: [
+      {
+        color: "#3498DB",
+        tooltip: "aaa",
+        value: 12,
+      },
+      {
+        color: "#E74C3C",
+        tooltip: "bbb",
+        value: 8,
+      },
+      {
+        color: "#F1C40F",
+        tooltip: "ccc",
+        value: 18,
+      },
+      {
+        color: "#2ECC71",
+        tooltip: "ddd",
+        value: 14,
+      },
+    ],
+  };
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("renders", () => {
+    render(<DoughnutChart {...defaultProps} />);
+    expect(screen.getByTestId("chart")).toBeInTheDocument();
+  });
+
+  it("displays the correct number of segments", () => {
+    render(<DoughnutChart {...defaultProps} />);
+    const segments = screen.getAllByTestId(TestIds.Segment);
+    expect(segments).toHaveLength(defaultProps.segments.length);
+  });
+
+  it("shows tooltips on hover", async () => {
+    render(<DoughnutChart {...defaultProps} />);
+    const firstSegment = screen.getAllByTestId(TestIds.Segment)[0];
+    await userEvent.hover(firstSegment);
+    await waitFor(() => {
+      expect(screen.getByRole("tooltip")).toBeInTheDocument();
+    });
+    // await waitFor(() => {
+    //     const tooltip = screen.getByTestId('tooltip-portal');
+    //     expect(tooltip).toBeInTheDocument();
+    //     expect(screen.getByRole('tooltip')).toHaveTextContent('aaa');
+    //   });
+
+    await userEvent.unhover(firstSegment);
+    await waitFor(() => {
+      expect(screen.queryByText("a")).not.toBeInTheDocument();
+    });
+  });
```

Comment:
I think you need `act` and maybe also the custom `userEventWithTimers` function for this to work. I.e. check this test from `Tooltip.test.tsx`

```
  it("updates the tooltip to fit on the screen", async () => {
    render(
      <Tooltip
        message="text that is too long to fit on the screen"
        position="right"
      >
        <button>Child</button>
      </Tooltip>
    );
    global.innerWidth = 20;
    await act(async () => {
      await userEventWithTimers.hover(
        screen.getByRole("button", { name: "Child" })
      );
      jest.runAllTimers();
    });
    expect(screen.getByTestId("tooltip-portal")).toHaveClass(
      "p-tooltip--btm-left"
    );
    expect(screen.getByTestId("tooltip-portal")).toHaveClass("is-detached");
  });
  ```

Reply:
Heya David,
I actually attempted to implement this last week to no avail. Something that is interesting is that the console prints out the html structure of the doughnut chart and when you analyse this, the tooltip test-ids do not show. I then figured it should be necessary to add another test-id in the opening span of the tooltip, however this broke another test in the Tooltip test suite regarding the code changing and no longer matching a previous snapshot. Therefore, as altering the Tooltip component is not an option, and accessing the Tooltip as a sub-component is not an option, I do not believe this can be done at this stage.

![image](https://github.com/user-attachments/assets/7d74cfb9-7cbe-4784-b829-23b38d646b9d)

The only thing that can be accessed from the section is the span which has a classname related to the tooltip (Which is also within the Tooltip component. I tried to put a test-id here), however as we cannot select by classname, this is inaccessible.

I wonder if the issue is about components not being registered as on hover, or whether Jest is simply not picking up on the Tooltip for one reason or another. In any case, I shall push my new (still failing) test for further review.

Reply:
This should work:
```

  it("shows tooltips on hover", async () => {
    const { container } = render(<DoughnutChart {...defaultProps} />);
    const segments = screen.getAllByTestId(TestIds.Segment);

    fireEvent.mouseOver(segments[0]);
    fireEvent.click(container.firstChild.firstChild);

    await waitFor(() => {
      expect(screen.getByText("aaa")).toBeInTheDocument();
    });
  });
  ```

Reply:
@edlerd , This works! Interesting to note that onHover actions do not register for this element. Question though, the selector container.firstChild.firstChild, doesn't that point to the <style> tag at line 153 in DoughnutChart.tsx as opposed to a Tooltip?
I will push the revised test suite now.